### PR TITLE
[editor] Check return value of m_getOriginalFeatureFn().

### PR DESCRIPTION
**Pull request by mgsergio**

### CODE REVIEW

File: indexer/osm_editor.cpp [hunk 1]
```diff
@@ -221,7 +221,20 @@ void Editor::LoadMapEdits()
           }
           else
           {
-            fti.m_feature = *m_getOriginalFeatureFn(fid);
+            auto const originalFeaturePtr = m_getOriginalFeatureFn(fid);
+            if (!originalFeaturePtr)
+            {
+              LOG(LERROR, ("A feature appears missing/deleted while loading edits."));
```
Comment (ygorshenin) [inline indexer/osm_editor.cpp]: s/appears/was/g

File: indexer/osm_editor.cpp [hunk 2]
```diff
@@ -449,12 +462,20 @@ Editor::SaveResult Editor::SaveEditedFeature(EditableMapObject const & emo)
   {
     ASSERT_NOT_EQUAL(featureStatus, FeatureStatus::Deleted, ("Unexpected feature status."));
 
+    auto const originalFeaturePtr = m_getOriginalFeatureFn(fid);
+    if (!originalFeaturePtr)
+    {
+      LOG(LERROR, ("A feature appears missing/deleted while saving."));
+      alohalytics::LogEvent("Editor_MissingFeature_Error");
```
Comment (ygorshenin) [inline indexer/osm_editor.cpp]: This is the first time I see this event - do we have similar events? Is it processed?

File: indexer/osm_editor.cpp [hunk 3]
```diff
@@ -614,8 +635,9 @@ EditableProperties Editor::GetEditableProperties(FeatureType const & feature) co
   // Disable opening hours editing if opening hours cannot be parsed.
   if (GetFeatureStatus(feature.GetID()) != FeatureStatus::Created)
   {
-    auto const & originalFeature = m_getOriginalFeatureFn(feature.GetID());
-    auto const & metadata = originalFeature->GetMetadata();
+    auto const originalFeaturePtr = m_getOriginalFeatureFn(feature.GetID());
+    ASSERT(originalFeaturePtr, ("The selected feature is missing some how."));
```
Comment (ygorshenin) [inline indexer/osm_editor.cpp]: Why assert? Why not proper handling of null ptr?

File: indexer/osm_editor.cpp (hunk 1 above)
Comment (ygorshenin) [inline indexer/osm_editor.cpp]: Feel free to leave this as is, but I'm against such long comments. This comment can be much simpler (with the same amount of information):

``` c++
LOG(LERROR, ("A feature can't be loaded:", fid));
```

File: indexer/osm_editor.cpp [hunk 4]
```diff
@@ -941,7 +985,10 @@ void Editor::MarkFeatureAsObsolete(FeatureID const & fid)
 
   auto & fti = m_features[fid.m_mwmId][fid.m_index];
   // If a feature was modified we can drop all changes since it's now obsolete.
-  fti.m_feature = *m_getOriginalFeatureFn(fid);
+  auto const originalFeaturePtr = m_getOriginalFeatureFn(fid);
+  ASSERT(originalFeaturePtr, ("The selected feature is missing some how."));
```
Comment (ygorshenin) [inline indexer/osm_editor.cpp]: Same here - we need to handle nullptr here too.

File: map/framework.cpp [hunk 5]
```diff
@@ -394,7 +394,8 @@ Framework::Framework()
   {
     unique_ptr<FeatureType> feature(new FeatureType());
     Index::FeaturesLoaderGuard const guard(m_model.GetIndex(), fid.m_mwmId);
-    guard.GetOriginalFeatureByIndex(fid.m_index, *feature);
+    if (!guard.GetOriginalFeatureByIndex(fid.m_index, *feature))
```
Comment (ygorshenin) [inline map/framework.cpp:397]: It's up to you, of course, but why not to modify the signature of the callback to:

``` c++
bool m_getOriginalFeatureFn(FeatureID const & id, FeatureType & ft);
```

?

File: indexer/osm_editor.cpp (hunk 2 above)
Comment (mgsergio) [inline indexer/osm_editor.cpp]: It's a new one.

File: indexer/osm_editor.cpp (hunk 1 above)
Comment (mgsergio) [inline indexer/osm_editor.cpp]: Done.

File: indexer/osm_editor.cpp (hunk 3 above)
Comment (mgsergio) [inline indexer/osm_editor.cpp]: Done.

File: indexer/osm_editor.cpp (hunk 4 above)
Comment (mgsergio) [inline indexer/osm_editor.cpp]: Done.

**Comment by mgsergio** [comment]
PTAL

**Comment by ygorshenin** [comment]
LGTM

**Pull request merged**